src/ydata_synthetic/synthesizers/gan.py

```diff
@@ -44,6 +44,8 @@ def __init__(
         self.noise_dim = model_parameters.noise_dim
         self.data_dim = model_parameters.n_cols
         self.layers_dim = model_parameters.layers_dim
+        self.cond_dim = model_parameters.condition
```

Comment:
This parameter does not belong to the base model. 

Please added it only in the class model where this attribute is required.